src/vignette/http/routes.clj

```diff
@@ -87,39 +99,47 @@
     (u/get-or-generate-thumbnail system (image-params->forced-thumb-params image-params))
     (get-original (store system) image-params)))
 
-; /lotr/3/35/Arwen.png/resize/10/10?debug=true
 (defn app-routes
   [system]
   (-> (routes
         (GET scale-to-width-route
              request
-             (image-request-handler system :thumbnail request
-                                    :thumbnail-mode "scale-to-width"
-                                    :height :auto))
+             (handle-thumbnail system
+                               (route->thumbnail-auto-height-map
+                                 (:route-params request)
+                                 request)))
         (GET window-crop-route
              request
-             (image-request-handler system :thumbnail request
-                                    :thumbnail-mode "window-crop"
-                                    :height :auto))
+             (handle-thumbnail system
+                               (route->thumbnail-auto-height-map
+                                 (:route-params request)
+                                 request)))
         (GET window-crop-fixed-route
              request
-             (image-request-handler system :thumbnail request
-                                    :thumbnail-mode "window-crop-fixed"
-                                    :height :auto))
+             (handle-thumbnail system
```

Comment:
Good catch. I think this should be `route->thumbnail-map` since the `:thumbnail-mode` should be in the `image-params` along with everything else that's in the URL. Fixed in b7dfbfa.